File: 프로그래머스/Python/0단계/짝수는 싫어요.py
# 정수 n이 매개변수로 주어질 때, n 이하의 홀수가 오름차순으로 담긴 배열을 return하도록 solution 함수를 완성하는 문제.

# 내 풀이

def solution(n):
    answer = []
    for i in range (n+1):
        if i%2 ==1:
            answer.append(i)
    return answer
  
  
  
 # 다른사람 풀이

def solution(n):
    return [i for i in range(1, n+1, 2)]

File: 프로그래머스/JavaScript/0단계/짝수는 싫어요.js
// n이 주워지면, n이하의 수 중 홀수만 저장한 배열을 출력하는 문제

// 내 풀이
function solution(n) {
    var arr = [];
    for(i=1;i <= n;i++){    //n이 중요하다..
        if (i%2===1){
            arr.push(i);
        }
    }
    return arr;
}


// 다른사람 풀이1
function solution(n) {
    var answer = [];

    for (let i = 1; i<=n; i+=2) answer.push(i)   // i+=2..

    return answer;
}

// 다른사람 풀이2(인덱스 활용)
function solution(n) {
    return Array(n).fill(1).map((v,i)=>v+i).filter(v=>v%2===1);
}
